### ISSUE TITLE
aggregateTimeout (kind of) ignored

### ISSUE THREAD

**Issue by AndreKR** [webpack/webpack-dev-server#1733]
- Operating System: Windows 7 Pro 64 bit
- Node Version: v10.15.0
- NPM Version: 6.7.0
- webpack Version: 4.29.6
- webpack-dev-server Version: 3.2.1

- [x] This is a **bug**

### Code

```js
// webpack.config.js
...
module.exports = function (env) {
	return [
		{
      ...
			devServer: {
				inline: true,
				hot: true,
				hotOnly: true,
			},
			watchOptions: {
				aggregateTimeout: 20000
			},
      ...
		}
	];
};
```

### For Bugs; How can we reproduce the behavior?

This is for Windows, I guess on Linux you can do something similar with `strace`.

* Add an `aggregateTimeout` of quite a few seconds (e.g. 20000) to your webpack config
* Run webpack-dev-server
* Run [Process Monitor](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon)
* In Process Monitor, filter for `Process Name = node.exe`
* Change one of the watched files

### Expected Behavior
When a file is changed, nothing should happen until the `aggregateTimeout` has elapsed. In particular, no files should be opened and locked.

### Actual Behavior
There is massive file activity, immediately after a file has been changed. Then there is a break of approximately the `aggregateTimeout`, then there is a little bit more file activity.

The actual problem is that immediately after a file is changed, the file will already be opened (=locked) by webpack-dev-server, ignoring the aggregateTimeout, making it impossible to, for example, do a `git rebase` while webpack-dev-server is running.

This behavior does *not* happen with `webpack --watch`, it only happens with webpack-dev-server.

**Comment by evilebottnawi** [comment]
`watchOptions` option only for `contentBase` feature, it is not affected on webpack

**Comment by evilebottnawi** [comment]
What is use case of `aggregateTimeout`? Because we migrate on built-in watcher in future major version

**Comment by AndreKR** [comment]
Well, in my case I want to be able to do a git rebase while webpack-dev-server is running.

Currently I cannot do that.

The moment that git changes a file, some files are opened and therefore locked by webpack-dev-server. Then, git wants to make more changes a few milliseconds later, but the files are already locked by webpack-dev-server, so the rebase operation fails with a "Permission denied" error.

**Comment by evilebottnawi** [comment]
@AndreKR it is very weird, i think doing this out of scope webpack-dev-server. Anyway you need setup `watchOptions` for webpack too https://webpack.js.org/configuration/watch/#watchoptions

**Comment by evilebottnawi** [comment]
hm we already respect `watchOptions`, please create minimum reproducible test repo, we need ivnestigate what is happens deeply

**Comment by AndreKR** [comment]
What do you mean? I already added `watchOptions: { aggregateTimeout: 20000 }` and for `webpack --watch` it works perfectly. But with webpack-dev-server the file is accessed immediately when it changes.

**Comment by AndreKR** [comment]
Ok, just a sec.

**Comment by AndreKR** [comment]
Ok, so here is the repro: https://github.com/AndreKR/issue-webpack-dev-server-1733

However, while setting this up I noticed that webpack watch also shows file activity immediately after a file change, yet I have no problems with locked files there.

I will investigate further to provide a more accurate report. Feel free to close the issue or leave it open in the meantime, I will report back when I have more information.

**Comment by evilebottnawi** [comment]
Feel free to feedback, let's keep open

**Comment by AndreKR** [comment]
It seemed to be a glitch that rebasing worked with `webpack --watch´ and currently I can't reproduce the issue anymore. Closing for now.
